.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: Changesets
+on:
+  workflow_dispatch:
+    inputs:
+      level:
+        description: 'Release type'
+        required: true
+        default: 'preview'
+        type: choice
+        options:
+          - preview
+          - general availability
+env:
+  CI: true
+  PNPM_CACHE_FOLDER: .pnpm-store
+jobs:
+  version:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: setup node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14
+      - name: install pnpm
+        run: npm i pnpm@latest -g
+      - name: Setup npmrc
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > .npmrc
+      - name: setup pnpm config
+        run: pnpm config set store-dir $PNPM_CACHE_FOLDER
+      - name: install dependencies
+        run: pnpm install
+      - name: set pre-release
+        if: ${{ inputs.level }} == 'preview'
+        run: pnpm changeset pre enter rc
+      - name: create and publish versions
+        uses: changesets/action@v1
+        with:
+          version: pnpm ci:version
+          commit: "chore: update versions"
+          title: "chore: update versions"
+          publish: pnpm ci:publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Is this needed? If so, how is this token defined please?

Reply:
This is a secret provided by github which allows actions to do actions against the repository, such as creating pull requests or pushing code. https://docs.github.com/en/actions/security-guides/automatic-token-authentication